clientapi/routing/directory.go

```diff
@@ -117,12 +117,14 @@ func SetLocalAlias(
 	// 1. The new method for checking for things matching an AS's namespace
```

Comment:
Perhaps it's time to put this in?

Reply:
Actually there doesn't seem to be an equivalent... There is [`IsInterestedInRoomAlias`](https://github.com/matrix-org/dendrite/blob/master/common/config/appservice.go#L103), but it doesn't check for `namespace.Exclusive`. Another reason for not doing it here is we should probably refactor this with all other occurrences in one batch.

Reply:
I meant that we could just use the method in conjunction with your `if device.UserID ...` check.

But yes, it would be nice to fix them all in one batch.